app/viewmodels/checkAnswers/SupplementaryUnitSummary.scala

```diff
@@ -17,24 +17,32 @@
 package viewmodels.checkAnswers
 
 import controllers.routes
-import models.{CheckMode, UserAnswers}
+import models.{CheckMode, RecordCategorisations, UserAnswers}
 import pages.SupplementaryUnitPage
 import play.api.i18n.Messages
+import queries.RecordCategorisationsQuery
 import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryListRow
 import viewmodels.govuk.summarylist._
 import viewmodels.implicits._
 
 object SupplementaryUnitSummary {
 
-  def row(answers: UserAnswers, recordId: String)(implicit messages: Messages): Option[SummaryListRow] =
-    answers.get(SupplementaryUnitPage(recordId)).map { answer =>
-      SummaryListRowViewModel(
-        key = "supplementaryUnit.checkYourAnswersLabel",
-        value = ValueViewModel(answer),
-        actions = Seq(
-          ActionItemViewModel("site.change", routes.SupplementaryUnitController.onPageLoad(CheckMode, recordId).url)
-            .withVisuallyHiddenText(messages("supplementaryUnit.change.hidden"))
-        )
-      )
+  def row(answers: UserAnswers, recordId: String)(implicit messages: Messages): Option[SummaryListRow] = {
+    val recordCategorisations = answers.get(RecordCategorisationsQuery).getOrElse(RecordCategorisations(Map.empty))
+    val categorisationInfoOpt = recordCategorisations.records.get(recordId)
+    categorisationInfoOpt match {
+      case Some(categorisationInfo) =>
```

Comment:
what do we do if the option is empty? Journey recovery?

Reply:
yeah I had that question in mind too, is it okay to redirect to journey recovery?

Reply:
I believe it should do that automatically but yes JourneyRecovery is where it should go.

Reply:
added a test case to make sure if it redirects journeyrecovery